core/src/main/java/org/openconnectors/connect/SinkConnector.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.openconnectors.connect;
+
+import org.openconnectors.config.Connector;
+import org.openconnectors.util.SinkConnectorContext;
+
+import java.util.Collection;
+
+public abstract class SinkConnector<T extends SinkConnectorContext, U> extends Connector<T> {
+
+    public abstract void publish(final Collection<U> messages) throws Exception;
```

Comment:
What about returning a `CompletableFuture` here instead of the flush?

Reply:
So suggestion is to wrap in a completable future and get rid of explicit flush method , how do other reviewers think about this.

Reply:
I like @merlimat suggestion of returning futures here. But feel that an eplicit flush might still be useful